pages/docs/reference/coding-conventions.md

```diff
@@ -726,6 +727,201 @@ foo {
+* [Parameters with optional type (including setters)](#parameters-with-optional-type-including-setters)
+* [Value arguments](#value-arguments)
+* [Indexing suffix](#indexing-suffix)
+* [Lambda parameters](#lambda-parameters)
+* [`when` entry](#when-entry)
+* [Collection literals (in annotations)](#collection-literals-in-annotations)
+
+#### Lists and enumerations
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+```kotlin
+val colors = listOf(
+    "red",
+    "green",
+    "blue", // trailing comma
+)
+
+enum class Direction {
+    NORTH,
+    SOUTH,
+    WEST,
+    EAST, // trailing comma
+}
+```
+
+</div>
+
+#### Class parameters
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+```kotlin
+class Customer(
+    val name: String,
+    val surname: String, // trailing comma
+)
+```
+
+</div>
+
+#### Function value parameters
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+```kotlin
+fun powerOf(
+    number: Int, 
+    exponent: Int, // trailing comma
+) { /*...*/ }
+
+constructor(
+    x: Comparable<Number>,
+    y: Iterable<Number>, // trailing comma
+) {}
+
+fun print(
+    vararg quantity: Int,
+    description: String, // trailing comma
+) {}
+```
+
+</div>
+
+#### Parameters with optional type (including setters)
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+```kotlin
+val foo: (Int, Int) -> Int = fun(
+    x,    
+    y, // trailing comma
+): Int {
+    return x + y
+}
+```
+
+</div>
+
+#### Value arguments
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+```kotlin
+fun sum(x: Any, y: Any) {}
+
+fun main() {
+    sum(
+        10,
+        20, // trailing comma
+    )
+}
+```
+
+</div>
+
+#### Indexing suffix
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+```kotlin
+class A {
+    operator fun get(x: Int, y: Int) = 10
+}
+
+fun foo(x: A) {
+    val y = x[
+            1,
+            3, // trailing comma
+    ]
+}
+```
+
+</div>
+
+#### Lambda parameters
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+```kotlin
+fun main() {
+    val x = {
+            x: Comparable<Comparable<Number>>,
+            y: Iterable<Iterable<Number>>, // trailing comma
+        ->
+            println("1")
+    }
+
+    println(x)
+}
+```
+
+</div>
+
+#### `when` entry
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+```kotlin
+fun foo(x: Any) = when (x) {
+    Comparable::class,
+    Iterable::class,
+    String::class, // trailing comma
+        -> println(1)
+    else -> println(3)
+}
+```
+
+</div>
+
+
+#### Collection literals (in annotations)
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+```kotlin
+annotation class Anno(val x: IntArray)
+
+@Anno([1, 2, 3, 4,])
```

Comment:
I believe the elements should be placed on separate lines otherwise the trailing comma loses all its advantages.